### PR TITLE
Fix for bug when providing superclass arguments as kwargs

### LINKED ISSUE #1068: Error when providing superclass arguments as kwargs
## 🐛 Bug

When some classes are instantiated with superclass arguments as kwargs, the superclass constructor receives multiple values for these arguments and instantiation fails with an error like:

`TypeError: torchmetrics.classification.stat_scores.StatScores.__init__() got multiple values for keyword argument 'reduce'`

This happens for the following classes:

- Accuracy
- Dice
- JaccardIndex
- Precision
- Recall
- Specificity

### To Reproduce

Instantiate Accuracy, Dice, Precision, Recall or Specificity specifying the argument `reduce` or `mdmc_reduce` as kwarg, or the class JaccardIndex with the `normalize` argument as kwarg.

#### Code sample

```
Accuracy(reduce='micro')
Accuracy(mdmc_reduce='global')
Dice(reduce='micro')
Dice(mdmc_reduce='global')
Precision(reduce='micro')
Precision(mdmc_reduce='global')
Recall(reduce='micro')
Recall(mdmc_reduce='global')
Specificity(reduce='micro')
Specificity(mdmc_reduce='global')
JaccardIndex(num_classes=1, normalize='true')
```

### Expected behavior

The class should be instantiated prioritizing values provided as kwargs, instead of default values generated inside the constructor.

### Environment

- TorchMetrics 0.9.0 built from source
- Python 3.10.4 & PyTorch 1.11.0
- WSL Ubuntu 20.04

### PR DESCRIPTION
**Pull request by jlcsilva**
## What does this PR do?

Fixes #1068

## Before submitting

- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/metrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to **update the docs**?
- [x] Did you write any new **necessary tests**?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/classification/test_accuracy.py [hunk 1]
```diff
@@ -448,3 +448,9 @@ def test_negmetric_noneavg(noneavg=_negmetric_noneavg):
     assert torch.allclose(noneavg["res1"], result1, equal_nan=True)
     result2 = acc(noneavg["pred2"], noneavg["target2"])
     assert torch.allclose(noneavg["res2"], result2, equal_nan=True)
+
+
+def test_provide_superclass_kwargs():
```
Comment (Borda) [inline tests/classification/test_accuracy.py:453]: let's move this to something like general classification tests and parametrize over classes

Comment (jlcsilva) [reply tests/classification/test_accuracy.py]: Moved it to the superclass test files, i.e., test_stat_scores and test_confusion_matrix. 
Any idea on how to make sure people build tests for this kind of situation going forward? Seems like a pretty easy thing to forget when adding new metrics classes or modifying the arguments of the current ones

**Review by Borda — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Comment by Borda** [comment]
Thank you! @jlcsilva mind checking the typing inconsistency flagged by `mypy`?
otherwise, we would like to include this in the next release this week :rabbit:

**Comment by jlcsilva** [comment]
Thanks everyone! Didn't have the time to check the typing inconsistency earlier, but your fix looks good to me @Borda

**Pull request merged**